File: problems/230B/230B.py
import math
input()
a = list(map(int, input().split()))


def solve(b):
    s = math.sqrt(b)
    s2 = math.sqrt(s)
    if int(s) != s or int(s2) == s2 or s % 2 == 0:
        return "NO"
    for i in range(3, math.floor(s2)+1, 2):
        if s % i == 0:
            return "NO"
    return "YES"


for i in a:
    if i == 1:
        print("NO")
    elif i == 4:
        print("YES")
    else:
        print(solve(i))

File: problems/230B/230B.cpp
#include <math.h>
#include <stdio.h>
#include <iostream>
void solve(long long b) {
  double s = sqrt(b);
  double s2 = sqrt(s);
  if (floor(s) != s || floor(s2) == s2 || (int)s % 2 == 0) {
    printf("1NO\n");
    return;
  }
  for (int i = 3; i < floor(s2) + 1;i += 2) {
    if (b % i == 0) {
      printf("2NO\n");
      return;
    }
  }
  printf("YES\n");
}
int main() {
  long long length;
  scanf("d", &length);

  int result[length];
  for (int i = 0; i < length;i++) {
    long long num;
    std::cin >> num;
    if (num == 1) {
      printf("NO\n");
    }
    else if (num == 4) {
      printf("YES\n");
    }
    else solve(num);
  }
}